File: frontend/src/types/tech_document.ts
export interface TechDocument {
  id: string
  section_id: string
  filename: string
  size_bytes: number
  version: number
  created_at: string
  created_by_user: {
    id: string
    full_name: string
  }
}

export interface TechDocumentVersion {
  id: string
  version: number
  filename: string
  created_at: string
  created_by_user: {
    id: string
    full_name: string
  }
}

export interface SheetPreview {
  name: string
  rows: string[][]
  total_rows: number
}

export interface TechDocumentPreview {
  filename: string
  sheets: SheetPreview[]
}

File: backend/app/schemas/tech_document.py
from datetime import datetime
from uuid import UUID
from typing import List

from pydantic import BaseModel

from app.schemas.user import UserSummary


class TechDocumentResponse(BaseModel):
    id: UUID
    section_id: UUID
    filename: str
    size_bytes: int
    version: int
    created_at: datetime
    created_by_user: UserSummary

    class Config:
        from_attributes = True


class TechDocumentUploadResponse(BaseModel):
    id: UUID
    filename: str
    version: int

    class Config:
        from_attributes = True


class TechDocumentVersionResponse(BaseModel):
    id: UUID
    version: int
    filename: str
    created_at: datetime
    created_by_user: UserSummary

    class Config:
        from_attributes = True


class SheetPreview(BaseModel):
    name: str
    rows: List[List[str]]
    total_rows: int

    class Config:
        from_attributes = True


class TechDocumentPreviewResponse(BaseModel):
    filename: str
    sheets: List[SheetPreview]

    class Config:
        from_attributes = True
